File: GXpert/GXpert.Web/Modules/ServerTypes/Exams/ExamQuestionColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { EDifficultyLevel } from "../Web/Enums.EDifficultyLevel";
import { ExamQuestionRow } from "./ExamQuestionRow";

export interface ExamQuestionColumns {
    Id: Column<ExamQuestionRow>;
    ExamTitle: Column<ExamQuestionRow>;
    ExamSectionTitle: Column<ExamQuestionRow>;
    QuestionText: Column<ExamQuestionRow>;
    EDifficultyLevel: Column<ExamQuestionRow>;
    RightAnswer: Column<ExamQuestionRow>;
    Marks: Column<ExamQuestionRow>;
    SortOrder: Column<ExamQuestionRow>;
    ClassTitle: Column<ExamQuestionRow>;
    SubjectTitle: Column<ExamQuestionRow>;
    TopicTitle: Column<ExamQuestionRow>;
}

export class ExamQuestionColumns extends ColumnsBase<ExamQuestionRow> {
    static readonly columnsKey = 'Exams.ExamQuestion';
    static readonly Fields = fieldsProxy<ExamQuestionColumns>();
}

[EDifficultyLevel]; // referenced types

File: GXpert/GXpert.Web/Modules/Exams/ExamQuestion/ExamQuestionColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Exams.Columns;

[ColumnsScript("Exams.ExamQuestion")]
[BasedOnRow(typeof(ExamQuestionRow), CheckNames = true)]
public class ExamQuestionColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string ExamTitle { get; set; }
    public string ExamSectionTitle { get; set; }
    public string QuestionText { get; set; }
    public short EDifficultyLevel { get; set; }
    [EditLink]
    public string RightAnswer { get; set; }
    public float Marks { get; set; }
    public float SortOrder { get; set; }
    public string CourseTitle { get; set; }
    public string ClassTitle { get; set; }
    public string SemesterTitle { get; set; }
    public string SubjectTitle { get; set; }
    public string TopicTitle { get; set; }
}